models/gc_spider_baseline/utils/SpiderBaselineRunner.py

```diff
@@ -0,0 +1,155 @@
+        # * The original image dimensions are used to generate the output segmentation
+        SpiderAlgorithm(
+            [
+                "SPIDER-Baseline",
+                "--epoch", "999999",
+                "--dataset", "spider_input",
+                "--surface_erosion_threshold", "-2000" if is_modality_mr else "200",
+                "--min_fragment_size", "100" if is_modality_mr else "500",
+                "--min_size_cont", "100" if is_modality_mr else "500",
+                "--export_original"
+            ]
+        ).execute()
+
+        source_output_file = self.SPIDER_INTERNAL_OUTPUT_DIR / "input_img_total_segmentation_original.mha"
+
+        self.log(f"Move raw generated segmentation output from: {source_output_file} -> {out_data_raw.abspath}", level="NOTICE")
+        shutil.move(str(source_output_file), out_data_raw.abspath)
+
+        # Create the remapping dictionary to reorder output labels so they will be picked up in the correct order by DicomSeg
+        remap_dict = {i:i for i in range(0, 26)}                    # keep labels 0-25 the same
+        remap_dict.update({i:i-100 for i in range(101, 126)})       # partially visible vertebrae get remapped to regular vertebrae labels
+        # TODO the following two lines can be added if their rois are added to the segdb
+        # remap_dict.update({i:i-201+26 for i in range(201, 226)})  # remaps intervertebral discs to 26-49
+        # remap_dict.update({100: 50})                              # remaps spinal canal to 50
+
+        # Convert the mapping to a 1d numpy vector by generating a value for each potential segmentation value
+        # Each value is mapped to zero by default and the mapping values are overwritten by the remap_dict
+        remap_np = np.zeros((226,), dtype=int)
+        remap_np[list(remap_dict.keys())] = list(remap_dict.values())
+
+        self.log(f"Remap generated segmentation output to: {out_data.abspath}", level="NOTICE")
+        self.log(f"  mapping used: {remap_dict}", level="DEBUG")
+        seg_sitk = SimpleITK.ReadImage(out_data_raw.abspath)
+        seg_np = SimpleITK.GetArrayFromImage(seg_sitk)
+        seg_remapped_np = remap_np[seg_np]  # actual remapping
+        seg_remapped_sitk = SimpleITK.GetImageFromArray(seg_remapped_np)
+        seg_remapped_sitk.CopyInformation(seg_sitk)
+        SimpleITK.WriteImage(seg_remapped_sitk, out_data.abspath, True)
+
+        # Run internal output cleanup
+        shutil.rmtree(str(self.SPIDER_INTERNAL_OUTPUT_DIR))
+
+    def create_arguments_config_file(self, traversal_direction_up: bool = True):
+        with open(self.SPIDER_EXPERIMENT_ARGUMENTS_FILE, "w") as f:
+            json.dump({
+                "filters": 50,  # Fixed for this model
+                "traversal_direction": "up" if traversal_direction_up else "down"
+                }, f
+            )
+
+    def create_input_metadata_file(self, modality: str, filetype: str = "mha"):
```

Comment:
Please explain what this is doing. References to the original algorithm's documentation can be helpful too.